src/FluxCalculator.jl

```diff
@@ -439,4 +383,95 @@ function water_albedo_from_atmosphere!(atmos_sim::Interfacer.AtmosModelSimulatio
+- `sim`: [Interfacer.ComponentModelSimulation] the surface simulation to compute fluxes for.
+- `atmos_sim`: [Interfacer.AtmosModelSimulation] the atmosphere simulation to compute fluxes with.
+- `boundary_space`: [CC.Spaces.AbstractSpace] the space of the coupler surface.
+- `thermo_params`: [TD.Parameters.ThermodynamicsParameters] the thermodynamic parameters.
+- `surface_scheme`: [AbstractSurfaceFluxScheme] the surface flux scheme.
+"""
+function compute_surface_fluxes!(
+    csf,
+    sim::Interfacer.AtmosModelSimulation,
+    atmos_sim::Interfacer.AtmosModelSimulation,
+    boundary_space,
+    thermo_params,
+    surface_scheme,
+)
+    # do nothing for atmos model
+    return nothing
+end
+
+function compute_surface_fluxes!(
+    csf,
+    sim::Interfacer.SurfaceModelSimulation,
+    atmos_sim::Interfacer.AtmosModelSimulation,
+    boundary_space,
+    thermo_params,
+    surface_scheme,
+)
+    # `_int` refers to atmos state of center level 1
+    z_int = Interfacer.get_field(atmos_sim, Val(:height_int))
+    uₕ_int = Interfacer.get_field(atmos_sim, Val(:uv_int))
+    thermo_state_int = Interfacer.get_field(atmos_sim, Val(:thermo_state_int))
+    z_sfc = Interfacer.get_field(atmos_sim, Val(:height_sfc))
+
+    # get area fraction (min = 0, max = 1)
+    area_fraction = Interfacer.get_field(sim, Val(:area_fraction))
+    # get area mask [0, 1], where area_mask = 1 if area_fraction > 0
+    area_mask = Utilities.binary_mask.(area_fraction)
+
+    thermo_state_sfc = FluxCalculator.surface_thermo_state(sim, thermo_params, thermo_state_int)
+
+    # set inputs based on whether the surface_scheme is `MoninObukhovScheme` or `BulkScheme`
+    surface_params = FluxCalculator.get_surface_params(atmos_sim)
+    scheme_properties = FluxCalculator.get_scheme_properties(surface_scheme, sim)
+
+    input_args = (; thermo_state_sfc, thermo_state_int, uₕ_int, z_int, z_sfc, scheme_properties, boundary_space)
+    inputs = FluxCalculator.surface_inputs(surface_scheme, input_args)
+
+    # calculate the surface fluxes
+    fluxes = FluxCalculator.get_surface_fluxes!(inputs, surface_params)
+    (; F_turb_ρτxz, F_turb_ρτyz, F_shf, F_lhf, F_turb_moisture) = fluxes
+
+    # perform additional diagnostics if required
+    FluxCalculator.differentiate_turbulent_fluxes!(sim, (thermo_params, input_args, fluxes))
+
+    # update fluxes in the coupler
+    fields = (;
+        F_turb_ρτxz = F_turb_ρτxz,
+        F_turb_ρτyz = F_turb_ρτyz,
+        F_turb_energy = F_shf .+ F_lhf,
+        F_turb_moisture = F_turb_moisture,
+    )
+
+    # update the fluxes of this surface model
+    FluxCalculator.update_turbulent_fluxes!(sim, fields)
+
+    # add the flux contributing from this surface to the coupler field
+    # note that the fluxes are area-weighted, so if a surface model is
+    #  not present at this point, the fluxes are zero
+    @. csf.F_turb_ρτxz += F_turb_ρτxz * area_fraction * area_mask
```

Comment:
do you also want to remove the area_mask here?

Reply:
I'll do this in a separate PR!